main.rb

```diff
@@ -0,0 +1,142 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    y = 0
+    arr = to_a
+    while y < arr.length
+      yield arr[y]
+      y += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    y = 0
+    arr = to_a
+    while y < arr.length
+      yield arr[y], y
+      y += 1
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    access = []
+    my_each do |val|
+      access << val if yield val
+    end
+    access
+  end
+
+  def my_all?
+    return (my_all? { |a| !a.nil? }) unless block_given?
+
+    my_each do |a|
+      return false unless yield a
+    end
+    true
+  end
```

Comment:
1. `my_all?` identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
...1. when no block or argument is given returns true when none of the collection members are false or nil.
```ruby
[1, 2, 3].my_all? #should return true.
```
...2. When a class is passed as an argument returns true if all of the collection is a member of such class. 
```ruby
[1, 2, 3].my_all?(Integer) #should return true
```
...3. When a Regex is passed as an argument returns true if all of the collection matches the Regex
```ruby
%w[dog door rod blade].my_all?(/d/) #should return true
```
...4. When a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
```ruby
[1, 1, 1].my_all?(1) #should return true
```